### ISSUE TITLE
small plots when rerunning the jupyter notebooks

### ISSUE THREAD

**Issue by blinkeye** [amueller/introduction_to_ml_with_python#36]
If the jupyter notebooks are executed (again) the resulting plots are a lot smaller than before. I suggest to set the _figure.figsize_ option in the **preample.py** to something like:

`plt.rcParams['figure.figsize'] = 15, 10`

**Comment by amueller** [comment]
Hey. Which version of Jupyter and matplotlib are you using?
I'm seeing DPI, which I feel is a better way to enlarge the figures. In
some versions of the notebook this setting is ignored sometimes.

**Comment by amueller** [comment]
The difference is that setting DPI also enlarges the font and markers
proportionally

**Comment by blinkeye** [comment]
I've downloaded the most recent `Anaconda3-4.4.0-Linux-x86_64.sh` version:
```
$ conda info
Current conda install:

               platform : linux-64
          conda version : 4.3.21
       conda is private : False
      conda-env version : 4.3.21
    conda-build version : not installed
         python version : 3.6.1.final.0
       requests version : 2.14.2
       root environment : /home/rglauser/anaconda3  (writable)
    default environment : /home/rglauser/anaconda3
       envs directories : /home/rglauser/anaconda3/envs
                          /home/rglauser/.conda/envs
          package cache : /home/rglauser/anaconda3/pkgs
                          /home/rglauser/.conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/free/linux-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/linux-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/linux-64
                          https://repo.continuum.io/pkgs/pro/noarch
            config file : None
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.21 requests/2.14.2 CPython/3.6.1 Linux/4.4.0-79-generic debian/stretch/sid glibc/2.23    
                UID:GID : 1000:1000

```

Matplotlib version:
```
$ conda list matplotlib
#
matplotlib                2.0.2               np112py36_0  
```

Jupyter version:
```
$ conda list jupyter
#
jupyter                   1.0.0                    py36_3  
jupyter_client            5.0.1                    py36_0  
jupyter_console           5.1.0                    py36_0  
jupyter_core              4.3.0                    py36_0 
```

**Comment by blinkeye** [comment]
I followed up on your hint about changing the DPI of the figures. That's a better solution, so I added

`plt.rcParams['figure.dpi'] = 150`

to **preample.py**

It looks to me like the default is _figure.dpi = 100_ which is a bit too small (and smaller than your initial figures).

**Comment by amueller** [comment]
Hmm I have to double check, it was 300 for the book, but maybe I changed it
later. Thanks for investigating! There was a version issue and I might have
not finished up the work on this.

**Comment by blinkeye** [comment]
I started with 300 (like you use for saving images in _preample.py_):
`plt.rcParams['savefig.dpi'] = 300`
but then the plots are quite a bit bigger than in the original notebook you commited.

**Comment by amueller** [comment]
hm no, it shouldn't be. And that's not for saving images, that's also for showing them in the notebook. That's why I was asking about your notebook version. Depending on which version of the notebook you have, this is used or not.

**Comment by blinkeye** [comment]
I posted the Matplotlib and Jupyter versions above. Do you mean the notebook version of your git repo? I'm using your master branch and the HEAD version, which is currently:
```
commit 409810c21060875d1c689713f6c42e0ce9cb2f1b (origin/master, origin/HEAD)
Merge: dbd8661 4f3c625
Author: Andreas Mueller <t3kcit@gmail.com>
Date:   Fri Jun 16 16:38:56 2017 +0200

    Merge pull request #43 from blinkeye/issue/37
    
    fix #37: display(panda DataFrame) throws a TypeError: 'module' object…

```
The notebooks all start with:
```
from preamble import *
%matplotlib inline
```

**Comment by amueller** [comment]
Oh sorry, overread the version.
My point was that ``plt.rcParams['savefig.dpi'] = 300`` should affect the plots generated inside the notebook.

**Comment by amueller** [comment]
What's
```
conda list notebook
```
?

**Comment by amueller** [comment]
I'm running
```
notebook                  4.2.3                    py35_0  
jupyter                   1.0.0                    py35_3  
jupyter_client            4.4.0                    py35_0  
jupyter_console           5.0.0                    py35_0  
jupyter_core              4.2.0                    py35_0  
```
and they are affected, will have to look into it more deeply, also check out https://github.com/jupyter/notebook/issues/2544

**Comment by amueller** [comment]
this is caused by:
https://github.com/ipython/ipykernel/pull/176
https://github.com/ipython/ipython/pull/9868

I'm replacing ``savefig`` with ``figure`` now, which should fix it in master in recent versions of ipykernel and ipython.

**Comment by amueller** [comment]
fixed in cccbbca86d00f9d5ffb643c740de4489de80436d
